src/Microsoft.CodeAnalysis.Razor.Workspaces/Host/IDocumentExcerptService.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if DOCUMENT_SERVICE_FACTORY
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    /// <summary>
+    /// excerpt some part of <see cref="Document"/>
```

Comment:
nit: Some of these doc comments could use a little polish

Reply:
This is a copy paste of work in progess code from Roslyn, these will be deleted as soon as possible.